r-package/grf/tests/testthat/test_local_linear_forest.R

```diff
@@ -233,3 +243,80 @@ test_that("output of tune local linear forest is consistent with prediction outp
   pred.ll.max <- predict(forest, ll.lambda = ll.max)$predictions
   expect_true(max(abs(tune.out$oob.predictions[, length(tune.out$lambdas)] - pred.ll.max)) < 10^-6)
 })
+
+test_that("local linear forests with local linear splits are numeric and include variance estimates", {
+  n <- 200
+  p <- 4
+
+  X <- matrix(runif(n * p, -1, 1), nrow = n)
+  X.test <- matrix(runif(n * p, -1, 1), nrow = n)
+  mu <- 0.9 * exp(X[, 1])
+  Y <- mu + rnorm(n)
+
+  forest <- ll_regression_forest(X, Y, num.trees = 800, ll.splits = TRUE, ci.group.size = 2)
+  preds.oob <- predict(forest, estimate.variance = TRUE)
+  preds.test <- predict(forest, X.test, estimate.variance = TRUE)
+
+  expect_true(is.numeric(preds.oob$predictions))
+  expect_true(is.numeric(preds.test$predictions))
+
+  expect_equal(n, length(preds.oob$variance.estimates))
+  expect_equal(n, length(preds.test$variance.estimates))
+})
+
+test_that("local linear splits reduce early splits on linear trends", {
+   n <- 200
+   p <- 6
+
+   X <- matrix(runif(n * p, -1, 1), nrow = n)
+   mu <- 0.9 * exp(X[, 1]) + 2 * X[,6]
+   Y <- mu + rnorm(n)
+
+   ll.forest <- ll_regression_forest(X, Y, ll.splits = TRUE)
+   ll.split.freq <- split_frequencies(ll.forest, 1)
+
+   forest = regression_forest(X, Y)
+   split.freq <- split_frequencies(forest, 1)
+
+   expect_true(split.freq[1,1] < ll.split.freq[1,1] / 2)
```

Comment:
Comment. I tightened this to
```suggestion
   expect_true(split.freq[1,1] < ll.split.freq[1,1] / 3)
```
and it still passed 100/100 times.